### PR TITLE
[core] Support custom data file name prefix

### PR DESCRIPTION
**Pull request by askwang**
### Purpose

Add conf `file.prefix` to control the file name prefix of data files.

### CODE REVIEW

**Comment by askwang** [comment]
help cc @JingsongLi  local repository tests success.

File: paimon-common/src/main/java/org/apache/paimon/CoreOptions.java [hunk 1]
```diff
@@ -171,6 +171,12 @@ public class CoreOptions implements Serializable {
                     .withDescription(
                             "Default file compression zstd level. For higher compression rates, it can be configured to 9, but the read and write speed will significantly decrease.");
 
+    public static final ConfigOption<String> FILE_PREFIX =
+            key("file.prefix")
```
Comment (JingsongLi) [inline paimon-common/src/main/java/org/apache/paimon/CoreOptions.java:175]: `data-file.prefix`

Default value is `"data-"`.

File: paimon-common/src/main/java/org/apache/paimon/CoreOptions.java [hunk 2]
```diff
@@ -171,6 +171,12 @@ public class CoreOptions implements Serializable {
                     .withDescription(
                             "Default file compression zstd level. For higher compression rates, it can be configured to 9, but the read and write speed will significantly decrease.");
 
+    public static final ConfigOption<String> FILE_PREFIX =
```
Comment (JingsongLi) [inline paimon-common/src/main/java/org/apache/paimon/CoreOptions.java:174]: Can you also introduce `changelog-file.prefix`?

Comment (askwang) [reply paimon-common/src/main/java/org/apache/paimon/CoreOptions.java]: done @JingsongLi

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**